common/remote/hooks/git/gitfile.cpp

```diff
@@ -410,10 +413,329 @@ class GitRepositoryDirectoryIterator : implements IDirectoryIterator, public CIn
+                if (git_oid_fromstrp(&gitOid, version) == 0)
+                {
+                    //User provided a SHA (possibly shorted) -> resolve it.  Error will be reported later if it does not exist.
+                    check(git_commit_lookup_prefix(&gitCommit, gitRepo, &gitOid, strlen(version)), "git_commit_lookup_prefix");
+                }
+            }
+
+            if (gitCommit)
+                check(git_commit_tree(&gitRoot, gitCommit), "git_commit_tree");
+        }
+    }
+    ~GitCommitTree()
+    {
+        git_tree_free(gitRoot);
+        git_commit_free(gitCommit);
+        git_repository_free(gitRepo);
+    }
+
+    const git_tree * queryTree() const { return gitRoot; }
+
+protected:
+    void check(int code, const char * func)
+    {
+        if (code != 0)
+        {
+            const git_error * err = git_error_last();
+            const char * errmsg = err ? err->message : "<unknown>";
+            WARNLOG("libgit %s returned %u: %s", func, code, errmsg);
+        }
+    }
+protected:
+    git_repository * gitRepo = nullptr;
+    git_commit * gitCommit = nullptr;
+    git_tree * gitRoot = nullptr;
+};
+
+
+class NewGitRepositoryFileIO : implements CSimpleInterfaceOf<IFileIO>
+{
+public:
+    NewGitRepositoryFileIO(const char * gitDirectory, const char * revision, const char * relFileName, GitCommitTree * commitTree, const git_oid * oid)
+    {
+        git_blob *blob = nullptr;
+        int error = git_blob_lookup(&blob, git_tree_owner(commitTree->queryTree()), oid);
+        if (error)
+            throw MakeStringException(0, "git git_blob_lookup returned exit status %d", error);
+
+        git_object_size_t blobsize = git_blob_rawsize(blob);
+        const void * data = git_blob_rawcontent(blob);
+        buf.append(blobsize, data);
+
+        git_blob_free(blob);
+    }
+    virtual size32_t read(offset_t pos, size32_t len, void * data)
+    {
+        if (pos >= buf.length())
+            return 0;
+        if (pos+len > buf.length())
+            len = buf.length()-pos;
+        memcpy_iflen(data, buf.toByteArray()+pos, len);
+        return len;
+    }
+    virtual offset_t size()
+    {
+        return buf.length();
+    }
+    virtual void close()
+    {
+    }
+
+    // Write methods not implemented - this is a read-only file
+    virtual size32_t write(offset_t pos, size32_t len, const void * data)
+    {
+        throwUnexpected();
+    }
+    virtual offset_t appendFile(IFile *file,offset_t pos=0,offset_t len=(offset_t)-1)
+    {
+        throwUnexpected();
+    }
+    virtual void setSize(offset_t size)
+    {
+        throwUnexpected();
+    }
+    virtual void flush()
+    {
+        throwUnexpected();
+    }
+    unsigned __int64 getStatistic(StatisticKind kind)
+    {
+        //This could be implemented, but not likely to be useful so currently return nothing.
+        return 0;
+    }
+protected:
+    MemoryBuffer buf;
+};
+
+class NewGitRepositoryFile : public GitRepositoryFile
```

Comment:
There is code around line 274 that creates a GitRepositoryFile object directly that may need chaging

Reply:
well spotted, I will change